Mathlib/LinearAlgebra/FreeModule/IdealQuotient.lean

```diff
@@ -107,3 +108,24 @@ noncomputable def Ideal.fintypeQuotientOfFreeOfNeBot [Module.Free ℤ S] [Module
     ⟨Int.natAbs_ne_zero.mpr (Ideal.smithCoeffs_ne_zero b I hI i)⟩
   classical exact Fintype.ofEquiv (∀ i, ZMod (a i).natAbs) e.symm
 #align ideal.fintype_quotient_of_free_of_ne_bot Ideal.fintypeQuotientOfFreeOfNeBot
+
+variable (F : Type _) [CommRing F] [Algebra F R] [Algebra F S] [IsScalarTower F R S]
+  (b : Basis ι R S) {I : Ideal S} (hI : I ≠ ⊥)
+
+/-- Decompose `S⧸I` as a direct sum of cyclic `R`-modules
+  (quotients by the ideals generated by Smith coefficients of `I`). -/
+noncomputable def Ideal.quotientEquivDirectSum :
+    (S ⧸ I) ≃ₗ[F] ⨁ i, R ⧸ Ideal.span ({I.smithCoeffs b hI i} : Set R) :=
+  ((I.quotientEquivPiSpan b _).restrictScalars F).trans
```

Comment:
```suggestion
  -- porting note: manual construction of `CompatibleSmul` typeclass no longer needed
  ((I.quotientEquivPiSpan b _).restrictScalars F).trans
```
since it's there in the mathport output